applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -425,24 +426,50 @@ namespace Kratos
 				- ReferenceCoordinates[i]);
 		}
 
-		//use orientation class 1st constructor
-		double theta_costum = 0.00;
-		if (this->GetProperties().Has(ANG_ROT)) theta_costum = this->GetProperties()[ANG_ROT];
-			
+		Matrix Temp = ZeroMatrix(msDimension);
+		this->mRotationMatrix0 = ZeroMatrix(msElementSize);
+
+		// take user defined local axis 2 from GID input
+		if (this->Has(LOCAL_AXIS_2)) 
+		{
+			double VectorNorm = MathUtils<double>::Norm(DirectionVectorX);
+			if (VectorNorm > numerical_limit) DirectionVectorX /= VectorNorm;
+
+			DirectionVectorY = this->GetValue(LOCAL_AXIS_2);
+			DirectionVectorZ[0] = DirectionVectorX[1]*DirectionVectorY[2]-DirectionVectorX[2]*DirectionVectorY[1];
+			DirectionVectorZ[1] = DirectionVectorX[2]*DirectionVectorY[0]-DirectionVectorX[0]*DirectionVectorY[2];
+			DirectionVectorZ[2] = DirectionVectorX[0]*DirectionVectorY[1]-DirectionVectorX[1]*DirectionVectorY[0];
+
+			VectorNorm = MathUtils<double>::Norm(DirectionVectorZ);
+			if (VectorNorm > numerical_limit) DirectionVectorZ /= VectorNorm;
+

```

Comment:
i think you should throw an error if the VectorNorm is zero

Reply:
here it could be done even in release since this function is called once in the whole calculation process

Reply:
Like that: 
LOCAL_AXIS_3 has length 0 !

?